src/lib/Inference.hs

```diff
@@ -2473,7 +2505,8 @@ instance Unifiable IxType where
 
 instance Unifiable DataDefParams where
   -- We ignore the dictionaries because we assume coherence
-  unifyZonked (DataDefParams xs _) (DataDefParams xs' _) = zipWithM_ unify xs xs'
+  unifyZonked (DataDefParams xs) (DataDefParams xs')
+    = zipWithM_ unify (leftsE xs) (leftsE xs')
```

Comment:
I am a little surprised that this still works, given that it actually does something different (see the comment). We should think hard and try to construct a case where this matters.

Reply:
I believe this actually does the same thing it used to do, i.e., still ignores the dictionaries.  The `leftsE` calls filter the dictionaries out.